matrix/Matrix.hpp

```diff
@@ -290,8 +290,8 @@ class Matrix
 
     void print() const
     {
-        char buf[200];
-        write_string(buf, 200);
+        char buf[1000];
```

Comment:
Ok, but if dynamic then using the fields for the exact size. Otherwise we have two disadavantages.